Userland/Libraries/LibRegex/RegexParser.h

```diff
@@ -53,6 +53,7 @@ class Parser {
         size_t match_length_minimum;
         Error error;
         Token error_token;
+        Vector<FlyString> capture_groups;
```

Comment:
Since we're now putting these in FlyStrings, let's make use of that as an optimisation and make the match state keys FlyStrings too